### PR TITLE
[6633] Remove all logical resources from code

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6633

### CODE REVIEW

**Review by rart — CHANGES_REQUESTED** [review]
Remove the useLogicResource and associated code too. Add to changelog

**Comment by jvega190** [comment]
@rart added back PreviewInPageInstancesPanel as it is being used by the PreviewComponentsPanel

File: ui/app/src/CHANGELOG.md [hunk 1]
```diff
@@ -36,10 +36,15 @@
...
 * [hooks]
   * `usePreviewUrlControl`: Removed `history` prop. Retrieval of search and navigate (previously called 'push') is now done internally.
+  * Removed `useLogicResource` hook.
+  * Removed `useSelectorResource` hook.
+  * Removed `useQuickCreateListResource` hook.
 * Removed deprecated `aws-file-upload` and `aws-video` upload controls.
 * Migrated the Studio UI build to Vite/SWC
 * Rollup's XB build to use SWC
 * Upgraded target compilation to ES2022, dropping many code transforms for features that are supported by most modern browsers such as nullish coalescing, optional chaining, object spreading and destructuring.
+* Removed `PreviewPageExplorerPanel` component.
```
Comment (rart) [inline ui/app/src/CHANGELOG.md]: This doesn't seem to have been removed? Nor should it.

**Review by rart — CHANGES_REQUESTED** [review]

**Comment by rart** [comment]
@jvega190 bump. this has pending changes.

File: ui/app/src/CHANGELOG.md (hunk 1 above)
Comment (jvega190) [reply ui/app/src/CHANGELOG.md]: this was incorrect, removing

**Review by rart — COMMENTED** [review]
- There's a `ComponentResource` Interface on PreviewBrowseComponentsPanelUI which is not in use. Try to find other resource-related things like this that might have also slipped by.

File: ui/app/src/components/ChangeContentTypeDialog/ChangeContentTypeDialogContainer.tsx [hunk 2]
```diff
@@ -61,8 +60,12 @@ export function ChangeContentTypeDialogContainer(props: ChangeContentTypeDialogC
   const [openSelector, setOpenSelector] = useState(false);
   const [selectedItem, setSelectedItem] = useState(item);
   const [contentTypes, setContentTypes] = useState<LegacyContentType[]>();
+  const [isFetching, setIsFetching] = useState(false);
   const [keyword, setKeyword] = useState('');
   const [debounceKeyword, setDebounceKeyword] = useState('');
+  const filteredContentTypes = contentTypes?.filter((contentType) =>
+    contentType.label.toLowerCase().includes(debounceKeyword.toLowerCase())
+  );
```
Comment (rart) [inline ui/app/src/components/ChangeContentTypeDialog/ChangeContentTypeDialogContainer.tsx]:
```suggestion
  const filteredContentTypes = useMemo(() => {
  let lowercaseKeyword = debounceKeyword.toLowerCase(); 
  return contentTypes?.filter((contentType) =>
    contentType.label.toLowerCase().includes(lowercaseKeyword)
  );
}, [contentTypes, debounceKeyword]);
```

File: ui/app/src/components/ChangeContentTypeDialog/ChangeContentTypeDialogContainer.tsx [hunk 3]
```diff
@@ -72,37 +75,25 @@ export function ChangeContentTypeDialogContainer(props: ChangeContentTypeDialogC
 
   useEffect(() => {
     if (selectedItem.path) {
-      fetchLegacyContentTypes(site, selectedItem.path).subscribe(
-        (response) => {
+      setIsFetching(true);
+      fetchLegacyContentTypes(site, selectedItem.path).subscribe({
```
Comment (rart) [inline ui/app/src/components/ChangeContentTypeDialog/ChangeContentTypeDialogContainer.tsx]: add cancellation

File: ui/app/src/components/ChangeContentTypeDialog/ChangeContentTypeDialogContainer.tsx [hunk 4]
```diff
@@ -138,32 +129,34 @@ export function ChangeContentTypeDialogContainer(props: ChangeContentTypeDialogC
...
+        ) : filteredContentTypes ? (
+          filteredContentTypes.length > 0 ? (
+            <Suspense fallback="">
+              <ContentTypesGrid
+                contentTypes={filteredContentTypes}
+                isCompact={isCompact}
+                onTypeOpen={onSelectedContentType}
+                selectedContentType={selectedContentType}
+              />
+            </Suspense>
```
Comment (rart) [inline ui/app/src/components/ChangeContentTypeDialog/ChangeContentTypeDialogContainer.tsx]: Is anything lazy/fetching/suspending here? What's the suspense for?

File: ui/app/src/components/CompareVersionsDialog/CompareVersionsDialogContainer.tsx [hunk 5]
```diff
@@ -138,9 +106,19 @@ export function CompareVersionsDialogContainer(props: CompareVersionsDialogConta
...
-          </SuspenseWithEmptyState>
+          compareVersionsBranch ? (
+            compareVersionsBranch.error ? (
+              <ApiResponseErrorState error={compareVersionsBranch.error} />
+            ) : compareVersionsBranch.isFetching ? (
+              <LoadingState />
+            ) : compareVersionsBranch.compareVersions?.length > 0 ? (
+              <Suspense fallback="">
+                <CompareVersions versions={compareVersionsBranch.compareVersions} />
+              </Suspense>
```
Comment (rart) [inline ui/app/src/components/CompareVersionsDialog/CompareVersionsDialogContainer.tsx]: Suspense doing anything?

Go over **all** of these. They should be only if there's a possibility of suspension.

File: ui/app/src/components/DeleteContentTypeDialog/DeleteContentTypeDialogContainer.tsx [hunk 6]
```diff
@@ -46,11 +47,31 @@ export function DeleteContentTypeDialogContainer(props: DeleteContentTypeDialogC
...
   });
+  const [{ data, isFetching, error }, setState] = useSpreadState({
+    data: null,
+    isFetching: false,
+    error: null
+  });
+
+  useEffect(() => {
+    setState({ isFetching: true });
+    fetchContentTypeUsage(site, contentType.id).subscribe({
```
Comment (rart) [inline ui/app/src/components/DeleteContentTypeDialog/DeleteContentTypeDialogContainer.tsx]: add cancellation

File: ui/app/src/components/DeletePluginDialog/UninstallPluginDialogContainer.tsx [hunk 7]
```diff
@@ -15,26 +15,46 @@
...
+    isFetching: false,
+    error: null
+  });
 
-  const resource = useMemo(() => {
-    return createResource(() => fetchMarketplacePluginUsage(site, pluginId).toPromise());
-  }, [site, pluginId]);
+  useEffect(() => {
+    setState({ isFetching: true });
+    fetchMarketplacePluginUsage(site, pluginId).subscribe({
```
Comment (rart) [inline ui/app/src/components/DeletePluginDialog/UninstallPluginDialogContainer.tsx]: add cancellation.

check **all** of these new effects on this PR for same deal...

File: ui/app/src/components/NewContentDialog/NewContentDialogContainer.tsx [hunk 8]
```diff
@@ -52,9 +51,15 @@ export function NewContentDialogContainer(props: NewContentDialogContainerProps)
...
   const [contentTypes, setContentTypes] = useState<LegacyContentType[]>();
+  const [isFetching, setIsFetching] = useState(false);
   const [keyword, setKeyword] = useState('');
   const [debounceKeyword, setDebounceKeyword] = useState('');
   const [selectedFilter, setSelectedFilter] = useState('all');
+  const filteredContentTypes = contentTypes?.filter(
+    (contentType) =>
+      contentType.label.toLowerCase().includes(debounceKeyword.toLowerCase()) &&
+      (selectedFilter === 'all' || contentType.type === selectedFilter)
+  );
```
Comment (rart) [inline ui/app/src/components/NewContentDialog/NewContentDialogContainer.tsx]: Same as above

File: ui/app/src/components/NewContentDialog/NewContentDialogContainer.tsx [hunk 9]
```diff
@@ -93,8 +98,10 @@ export function NewContentDialogContainer(props: NewContentDialogContainerProps)
           ? `${selectedItem.path}/`
           : selectedItem.path;
 
+      setIsFetching(true);
       fetchLegacyContentTypes(site, path).subscribe({
```
Comment (rart) [inline ui/app/src/components/NewContentDialog/NewContentDialogContainer.tsx]: cancellation?

File: ui/app/src/components/PreviewInPageInstancesPanel/PreviewInPageInstancesPanel.tsx [hunk 10]
```diff
@@ -95,6 +93,8 @@ export function PreviewInPageInstancesPanel() {
   const models = useMemo(() => {
     return guest?.models;
   }, [guest]);
+  const filteredContentTypes =
+    models && Object.values(models)?.filter((model) => model.craftercms.contentTypeId === contentTypeFilter);
```
Comment (rart) [inline ui/app/src/components/PreviewInPageInstancesPanel/PreviewInPageInstancesPanel.tsx]: I don't think Object.values would ever result in null.
```suggestion
    Object.values(models ?? {}).filter((model) => model.craftercms.contentTypeId === contentTypeFilter);
```

File: ui/app/src/components/PreviewInPageInstancesPanel/index.ts [hunk 11]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2007-2022 Crafter Software Corporation. All Rights Reserved.
+ * Copyright (C) 2007-2024 Crafter Software Corporation. All Rights Reserved.
```
Comment (rart) [inline ui/app/src/components/PreviewInPageInstancesPanel/index.ts]: File has no modifications. Roll this back here.

File: ui/app/src/components/ToolsPanel/ToolsPanel.tsx [hunk 12]
```diff
@@ -106,46 +96,46 @@ export function ToolsPanel() {
...
 
 function ToolsPaneBody(props: ToolsPaneBodyProps) {
-  const root = props.resource.read();
+  const { root, pageStack } = props;
   const site = useActiveSiteId();
-  const { pageStack } = props;
   const { rolesBySite } = useActiveUser();
   return (
-    <>
+    <Suspense fallback="">
```
Comment (rart) [inline ui/app/src/components/ToolsPanel/ToolsPanel.tsx]: This suspense definitely makes sense

**Review by rart — CHANGES_REQUESTED** [review]

**Comment by jvega190** [comment]
> * There's a `ComponentResource` Interface on PreviewBrowseComponentsPanelUI which is not in use. Try to find other resource-related things like this that might have also slipped by.

Should I get rid of this signature too?  https://github.com/craftercms/studio-ui/blob/develop/ui/app/src/components/StoreProvider/StoreProvider.tsx#L26

**Review by rart — APPROVED** [review]

**Comment by rart** [comment]
> > * There's a `ComponentResource` Interface on PreviewBrowseComponentsPanelUI which is not in use. Try to find other resource-related things like this that might have also slipped by.
> 
> Should I get rid of this signature too? https://github.com/craftercms/studio-ui/blob/develop/ui/app/src/components/StoreProvider/StoreProvider.tsx#L26

Yes. Why wouldn't we?

**Comment by rart** [comment]
Pls fix conflicts.

**Comment by jvega190** [comment]
@rart there were still a couple of resource-related things I removed, I'll re-request review

File: ui/app/src/utils/resource.ts
Comment (rart) [inline ui/app/src/utils/resource.ts:1]: Roll this one back, will be used elsewhere.

File: ui/app/src/models/Resource.ts
Comment (rart) [inline ui/app/src/models/Resource.ts:1]: Rollback

File: ui/app/src/models/index.ts [hunk 13]
```diff
@@ -55,7 +55,6 @@ export * from './PluginFileBuilder';
 export * from './ProjectLifecycleEvent';
 export * from './Publishing';
 export * from './Repository';
-export * from './Resource';
```
Comment (rart) [inline ui/app/src/models/index.ts]: Rollback

File: ui/app/src/components/PreviewDropTargetsPanel/PreviewDropTargetsPanel.tsx [hunk 14]
```diff
@@ -108,6 +105,13 @@ export function PreviewDropTargetsPanel() {
     });
     return allowedTypes;
   }, [allowedTypesData, contentTypes]);
+  const filteredDropTargets = dropTargetsBranch
+    ? dropTargetsBranch.byId
+      ? Object.values(dropTargetsBranch.byId).filter(
+          (dropTarget) => dropTarget.contentTypeId === dropTargetsBranch.selectedContentType
+        )
+      : []
+    : null;
```
Comment (rart) [inline ui/app/src/components/PreviewDropTargetsPanel/PreviewDropTargetsPanel.tsx]: add memo

File: ui/app/src/CHANGELOG.md
Comment (rart) [inline ui/app/src/CHANGELOG.md:1]: These changes move to 5.0

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**